module/wishlist/helper.js

```diff
@@ -0,0 +1,185 @@
+			let userExtensionDocument = await userExtensionQueries.userExtensionDocument({ userId })
+
+			let updateuserExtensionDocument
+			if (userExtensionDocument && userExtensionDocument.length > 0) {
+				// If the document exists, update the wishlist by appending the new item
+				updateuserExtensionDocument = await userExtensionQueries.findAndUpdate(
+					{ userId },
+					{ $addToSet: { wishlist: wishlistItem } }
+				)
+			} else {
+				// If no document exists, create a new one
+				const newUserExtension = {
+					userId,
+					wishlist: [wishlistItem],
+				}
+				updateuserExtensionDocument = await userExtensionQueries.createUserExtension(newUserExtension)
+			}
+			if (updateuserExtensionDocument) {
+				return {
+					success: true,
+					message: CONSTANTS.apiResponses.WISHLIST_ADDED,
+				}
+			} else {
+				throw {
+					message: CONSTANTS.apiResponses.WISHLIST_NOT_ADDED,
+					status: HTTP_STATUS_CODE.bad_request.status,
+				}
+			}
+		} catch (error) {
+			return {
+				status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+				message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+				errorObject: error,
+			}
+		}
+	}
+
+	/**
+	 * remove wishlist
+	 * @method
+	 * @name remove
+	 * @param {projectTempleteId} - pprojectTempleteId
+	 * @param {userId} -            userId
+	 * @returns {Object} .
+	 */
+	static async remove(projectTempleteId, userId) {
+		try {
+			// Find the userExtension document for the given userId
+			let userExtensionDocument = await userExtensionQueries.userExtensionDocument({ userId })
+
+			if (userExtensionDocument) {
+				// If the document exists, update the wishlist by removing the specified item
+				let updatedDocument = await userExtensionQueries.findAndUpdate(
+					{ userId, wishlist: { $elemMatch: { _id: projectTempleteId } } },
+					{ $pull: { wishlist: { _id: projectTempleteId } } },
+					{ new: true }
+				)
+				if (updatedDocument) {
+					return {
+						success: true,
+						message: CONSTANTS.apiResponses.WISHLIST_REMOVED,
+					}
+				} else {
+					throw {
+						message: CONSTANTS.apiResponses.PROJECTTEMPLATES_NOTFOUND,
+						status: HTTP_STATUS_CODE.bad_request.status,
+					}
+				}
+			} else {
+				throw {
+					message: CONSTANTS.apiResponses.USEREXTENSION_NOTFOUND,
+					status: HTTP_STATUS_CODE.bad_request.status,
+				}
+			}
+		} catch (error) {
+			return {
+				status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+				message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+				errorObject: error,
+			}
+		}
+	}
+
+	/**
+	 * list wishlist
+	 * @method
+	 * @name list
+	 * @param {String} language  -languageCode
+	 * @param {String}  userId   -userId
+	 * @param {Number} pageSize  -pageSize
+	 * @param {Number} pageNo    -pageNo
+	 * @returns {Object}          list of wishlist
+	 */
+
+	static async list(language, userId, pageSize = 100, pageNo = 1) {
```

Comment:
@praveenKDass what is the need to hard code pagesize and page number. if did't pass it by default it will 100 and 1 . you can verify once by consoling them in controller without passing any value during the api call. we won't need this

Reply:
Yes removing default pageSize and pageno @VISHNUDAS-tunerlabs